### PR TITLE
Initial project isolation support

### PR DESCRIPTION
**Pull request by modmuss50**
This PR disables some none essential features to allow Loom to run with project isolation.

What doesnt work with project isolation:
- Applying AWs or TAWs from project dependencies that are mdos.
- Reading any Gradle property.
  - https://github.com/gradle/gradle/issues/28857
  - Or a custom property parser!? surely it cannot be the only solution.
- Sandbox support (doesnt matter, was mostly to help me test stuff)
- Remap jar with other projects mixin AP mappings. (An edge case that I might disable by default anyway)
  - Using TR to remap mixins solves this problem.

The multi mc version test has been rewritten to run with project isolation and configure on demand.

### CODE REVIEW

File: src/test/resources/projects/multi-mc-versions/settings.gradle [hunk 1]
```diff
@@ -1,3 +1,14 @@
+plugins {
+	id 'fabric-loom' apply false
+}
+
+gradle.lifecycle.beforeProject {
+	if (it.rootProject == it) return
+
+		apply plugin: 'fabric-loom'
```
Comment (Juuxel) [inline src/test/resources/projects/multi-mc-versions/settings.gradle]:
```suggestion
	apply plugin: 'fabric-loom'
```
(to match the next line)

**Review by Juuxel — APPROVED** [review]
